Egardia redesign - generic component and sensor support

jeroenterheerdt opened the pull request:
## Description:
We re-architected the alarm_control_panel.egardia component to be a generic component. Also, adding in sensor support. Docs not yet written, code is not high quality yet: works but needs some review! Will write docs after code is in good shape.

**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#<home-assistant.github.io PR number goes here>

## Example entry for `configuration.yaml` (if applicable):
```yaml
egardia:
    host: YOUR_HOST
    username: YOUR_USERNAME
    password: YOUR_PASSWORD
```

## Checklist:
  - [X] The code change is tested and works locally.

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [X] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [X] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [X] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [X] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [X] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

jeroenterheerdt committed:
Adding Egardia sensor support.

jeroenterheerdt committed:
Adding Egardia sensor support.

jeroenterheerdt committed:
Adding Egardia sensor support.

jeroenterheerdt committed:
Adding Egardia sensor support.

jeroenterheerdt committed:
Fixing linting errors.

jeroenterheerdt committed:
Fixing linting errors.

jeroenterheerdt commented:
Travis build fails because `requirements_all.py` is not up to date. Running `gen_requirements_all.py` returns and error cannot import name `PROJECT_NAME` from homeassistant.const.... now what?

pvizeli committed:
Hass.io panel v2 (#11860)

* Update hassio.py

* fix test

* Update test_hassio.py

sdague committed:
Add missing mock call for test_minimal_config (#11858)

In local testing py36 always fails on test_minimal_config on a wait
call that never completes. One difference between this test and others
in the class is the lack of a mock on `call`. With this added, the
test passes locally 100%.

bitglue committed:
Expose Alexa Smart Home via HTTP POST (#11859)

* Expose Alexa Smart Home via HTTP POST

Haaska uses the deprecated v2 Alexa Smart Home payload. Exposing the v3
implementation this way allows an easy path to upgrading Haaska and
reducing code duplication with Home Assistant Cloud.

* Expose Alexa Smart Home via HTTP POST

Haaska uses the deprecated v2 Alexa Smart Home payload. Exposing the v3
implementation this way allows an easy path to upgrading Haaska and
reducing code duplication with Home Assistant Cloud.

PaulAnnekov committed:
fixed async function called in a thread (#11882)

nkgilley committed:
upgrade ecobee (#11878)

Rendili committed:
Fix for hive aux_heat target temperature display (#11884)

sdague committed:
Bump waterfurnace library to 0.3.0 (#11888)

Fixes in the library to handle transaction id which has to roll over
otherwise the server gets confused.

amelchio committed:
Postpone "Z-wave not ready" warning (#11807)

* Release worker thread while waiting for Z-wave startup

* Increase zwave startup timeout

* Adjust test

* Use asyncio.sleep in _check_awaked

* Remove lint

* Name loop parameter

rofrantz committed:
[SMALL-PR] Don't make climate calls if feature is not supported (#11841)

* Don't make climate calls if feature is not supported

* Implemented suggested code review changes & cleanup

* Fix demo climate unit tests

rofrantz committed:
Handle Daikin AC adapters without fan mode and swing mode support (#11840)

* Handle daikin AC devices without FAN_MODE and SWING_MODE support

* Handle daikin AC devices without FAN_MODE and SWING_MODE support

jeroenterheerdt commented:
@MartinHjelmare do you happen to know what might cause this issue?

GregoryDosh committed:
device tracker - tomato https support (#11566)

* initial https support

* adding tests

* lint errors

* missing docstring

* fixing non-deterministic params

* fixing non-deterministic params

* Updating docstrings & added missing tests

* revert _LOGGER

* updating default port to reflect ssl/nonssl

* fixing docstrings for tests

arcsur committed:
Bme680 (#11892)

* Added documentation reference to code

* Some code clean-ups suggested by @MartinHjelmare in PR#: 11695 after merging.

* fixed minor typo in docstring

* fixed another minor typo in same docstring...

* fixed another minor typo in same docstring...

MartinHjelmare commented:
Not sure. You seem to be up to date with commit history. Your branch looks OK. I'd try running `script/setup` script. If that doesn't solve it, try uninstalling homeassistant and run the script again.

```
pip3 uninstall homeassistant
script/setup
```

mu4yu3 committed:
Add android option for linux_battery.py (#11852)

* Add android option for linux_battery.py

* Add system option for linux_battery.py, support android.

todschmidt committed:
Pushbullet (fix) invalid keyword, added unittests (#11804)

* Fixed unittests

* revert requirements

* added pushbullet.py to requirements

* updated requirements

todschmidt committed:
Snips - (fix) removed endSession mqtt response on error and unknown intents (#11908)

* Remove snips endSession response on unknownIntent

* Removed snips_response for unknown and error.

balloob committed:
Clarify emulated hue warning (#11910)

balloob committed:
Use API to discover Hue if no bridges specified (#11909)

* Use API to discover Hue if no bridges specified

* hide config file

Commit:
Update python-wink version and multiple wink fixes/updates. (#11833)

Cinntax committed:
New venstar climate component (#11639)

* New venstar climate component

* Corrected items raised during the PR process.

* Corrected flak8 issues.

* Add support for venstar models without humidity control.

* Changed the idle operation mode to off.  Few other adjustments.

* Minor changes (log messages, sorting, etc.)

jinnerbichler committed:
Iota wallet (#11398)

* worked on first version of IOTA wallet

* added sensor for IRI node

* added requirements

* ignored new component in test coverage

* changed length of line

* fixed linter issues

* changed requested changes

* moved attributes of IRI node

* change coverage definitions

* changed loading of platforms

* changed loading of platforms

* changed initial state of wallet sensor

* changed handling of config

* changed attribute handling for IOTA node sensor

* Follow the component style

* Fix ordering and docstring

* Fix pylint issue

i-am-shodan committed:
Improve foscam library exception support (#11701)

* Improve foscam library exception support

Catches foscam exceptions that otherwise pollute the log. Many of these exception can safely be ignored

* Fixed line length

Fixed line length

* Changed exception and log handling

changed logging and catched only the TypeError effecting the library

* Removed unused var

Removed var

* Fix remaining issue

jeradM committed:
Added Xeoma camera platform (#11619)

* Added Xeoma camera platform

* Switch to dict.get() for optional config values with no default

* Minor changes

dfiel committed:
Multi-Room Support for Greenwave Reality (#11364)

* Multi-Room Support for Greenwave Reality

* PEP8 Line Too Long

* Update requirements_all.txt

* Shared State Object

Shared State Object added, and implemented new function of greenwavereality to abstract complex runtimes.

* Update requirements_all.txt

* Lint issues

* Rewrite method to not trigger "Method could be a function"

* Remove unnecessary arguments for update call

arcsur committed:
Minor fix to configuration validation and related log line. (#11898)

bitglue committed:
Report scripts and groups as scenes to Alexa (#11900)

* Send Alexa Smart Home responses to debug log

* Report scripts and groups as scenes to Alexa

The Alexa API docs have a couple display categories that sound relevant
to scenes or scripts:

    ACTIVITY_TRIGGER: Describes a combination of devices set to a
    specific state, when the state change must occur in a specific
    order.  For example, a “watch Neflix” scene might require the: 1. TV
    to be powered on & 2. Input set to HDMI1.

    SCENE_TRIGGER: Describes a combination of devices set to a specific
    state, when the order of the state change is not important. For
    example a bedtime scene might include turning off lights and
    lowering the thermostat, but the order is unimportant.

Additionally, Alexa has a notion of scenes that support deactivation.
This is a natural fit for groups, and scripts with delays which can be
cancelled.

https://developer.amazon.com/docs/device-apis/alexa-discovery.html#display-categories

The mechanism to map entities to the Alexa Discovery response is
refactored since extending the data structures in MAPPING_COMPONENT to
implement supportsDeactivation would have added complication to what I
already found to be a confusing construct.

nkgilley committed:
Add ERC20 tokens to etherscan.io sensor (#11916)

* upgrade python-etherscan-api

* add support for erc20 tokens

nkaminski committed:
Allow separate command and state OIDs and payloads in SNMP switch (#11075)

* Allow separate command and state OIDs and payloads in SNMP switch

Follow PEP8 style properly

More PEP8 style fixes

Fully PEP8 compliant

* Fix pylint errors

* Remove comment to satisfy pylint

* Style changes/refactoring

balloob committed:
Update frontend to 20180126.0

balloob committed:
Allow setting climate devices to AUTO mode via Google Assistant (#11923)

* Allow setting climate devices to AUTO mode via Google Assistant

* Remove cast to lower

* Clarify const name

andrey-git committed:
Add "write" service to system_log (#11901)

* Add API to write error log

* Move write_error api to system_log.write service call

* Restore empty line

akloeckner committed:
Map media_stop to idle state (#11819)

adresses #11813

basschipper committed:
Fixed rfxtrx binary_sensor KeyError on missing optional device_class (#11925)

* Fixed rfxtrx binary_sensor KeyError on missing optional device_class

* Fixed rfxtrx binary_sensor KeyError on missing optional device_class

bachya committed:
Adds allergy/disease sensor platform from Pollen.com (#11573)

* Base platform in place

* Logic in place

* Requirements and coverage

* Fixed some linting issues

* Small attribute reorganization

* Collaborator-requested changes round 1

* Updated documentation

kennedyshead committed:
fixes #11848 (#11915)

Adding tests to check the component after latest patch

bitglue committed:
Implement Alexa temperature sensors (#11930)

dotlambda committed:
check_config.py: allow colorlog==3.1. (#11927)

danielperna84 committed:
Update pyhomematic to 0.1.38 (#11936)

ReneNulschDE committed:
Weblink - Allow relative urls in config (#11808)

* Allow relative url

* Allow absolute urls in config schema

* change after pylint build

* Add tests and change error message

* Change regex to check starting forward slash only

* Change error message and const name

ReneNulschDE committed:
Panel_Iframe - Allow relative urls in config (#11832)

* Panel_Iframe - Allow relative urls in config

* change regex to check for starting forward slash only

* Change error message and const name

jeroenterheerdt committed:
Merge branch 'egardia' of https://github.com/jeroenterheerdt/home-assistant into egardia

jeroenterheerdt committed:
Updating requirements_all.txt

jeroenterheerdt commented:
@MartinHjelmare running setup resolved the issue, thanks!

jeroenterheerdt commented:
I know. I rebased  and they still showed up. What do I do now? Delete PR and create a new one?

jeroenterheerdt commented:
Replaced by #11994
